src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1956,6 +1956,17 @@ public unsafe string TrimStart(params char[]? trimChars)
             }
         }
 
+        public string RemoveStart(string value) => RemoveStart(value, StringComparison.Ordinal);
+
+        public unsafe string RemoveStart(string value, StringComparison comparisonType)
+        {
+            if (value == null)
+                throw new ArgumentNullException(nameof(value));
```

Comment:
nit: like other places in this file, we can use helper to avoid allocations:
```suggestion
                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.value);
```

Reply:
Thanks for your advice, it does avoid redundant allocations！

Reply:
It doesn't, instead it can help prevent cold blocks being inlined.

Reply:
> > it does avoid redundant allocation
> 
> It doesn't, instead it helps prevent cold blocks being inlined.

Thanks for correcting my mistake, but I don't quite understand how does it help prevent cold blocks inlined?

Reply:
Throws are always being considered as cold blocks by JIT, and a code path containing throws are generally not being inlined. Moving throws from a method to another helper method not only make the method itself possible to be inlined, but also helps shrink the generated code size.

See differences here: https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKBuIGYACMxgYRoG8bGfmniUjALIwMACwgATAOoBLcQBFZsMBgAqYqBADuACgCU3Xl2q8zzcgE5dAIWySAggAcnW7GDG6ArPv0BuRgB6QMZISRhGJ2xxRm15CQBXDEZxLW1cI3MLaztHFzcPXRRfAODQqQiomLiY1J0M014AX0zGVoZmQRFxKTlxAEkAO0llGFUNNINWkyyeYitdAHFYaOdXCHdPH38gkLDK6LFY+IgklM161rN562WYVfyNwuKdsv3Iw+Pai/TWlsaeO1BBMdAZGBxzmlGIMYNpGABRBBgGBODCyCCDAwBf5mVqyQbJXJrAqefHJBCGAHgq68WQAM0YugQjAAvCzGD5mAB2Dl+Gk8GAAG1wETqcJhcMRyNR6Mx/j+rTxBMYdwe602ujJjAp035jHpjOZbI5+m5vL1QpFjBBenlVOuPPCdOwCUFGFKIW0GIA5MkVh5GGIYLAFdR/kA===

Reply:
> Throws are always being considered as cold blocks by JIT, and a code path containing throws are generally not being inlined. Moving throws from a method to another helper method not only make the method itself possible to be inlined, but also helps shrink the generated code size.
> 
> See differences here (`MethodWithDirectThrow` and `MethodWithIndirectThrow`): https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKBuIGYACMxgYRoG8bGfmniUjALIwMACwgATAOoBLcQBFZsMBgAqYqBADuACgCU3Xl2q8zzcgE5dAIWySAggAcnW7GDG6ArPv0BuRgB6QMZISRhGJ2xxRm15CQBXDEZxLW1cI3MLaztHFzcPXRRfAODQqQiomLiY1J0M014AX0zGVoZmQRFxKTlxAEkAO0llGFUNNINWkyyeYitdAHFYaOdXCHdPH38gkLDK6LFY+IgklM161rN562WYVfyNwuKdsv3Iw+Pai/TWlsaeO1BBMdAZGBxzmlGIMYNpGABRBBgGBODCyCCDAwBf5mVqyQbJXJrAqefHJBCGAHgq68WQAM0YugQjAAvCzGD5mAB2Dl+Gk8GAAG1wETqcJhcMRyNR6Mx/j+rTxBMYdwe602ujJjAp035jHpjOZbI5+m5vL1QpFjBBenlVOuPPCdOwCUFGFKIW0GIA5MkVh5GGIYLAFdR/kA===


I see.Thank you very much for your detailed explanation!